### PR TITLE
UHF-3212: Add Journey Planner

### PR DESCRIPTION
**Pull request by xkhaven**
# [UHF-3212](https://helsinkisolutionoffice.atlassian.net/browse/UHF-3212)

## What was done

* Added new paragraph to embed Journey Planner (Reittiopas) on Page
* Added cookie compliance check

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-3212_Journey_planner_embbed`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-3212_Journey_planner_embbed`
* Run `make drush-cr`

## How to test

* [x] Paragraph "Journey Planner" is enabled for Page content type on "Upper content region"
* [x] Paragraph has fields Title and Description, with FI translation
* [x] The planner won't be loaded if cookies are not accepted
* [x] With cookies accepted the planner is loaded
* [x] The planner has working accessibility via TAB key as navigating above and below of the content, when it is loaded.
* [x] Translations for for the planner, accessibility functionality and iframe `label` attribute are correct
* [x] Content has responsive layout
* [x] Check that code follows our standards

## Designers review

* [X] This PR does not need designers review

## Other PRs

* [UHF-3212: Journey planner cookie compliance support](https://github.com/City-of-Helsinki/drupal-hdbt/pull/697)

[UHF-3212]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-3212?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: conf/cmi/paragraphs.paragraphs_type.journey_planner.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+uuid: 90392db0-a1b5-4348-a2b9-c672d59d6f3d
```
Comment (tuutti) [inline conf/cmi/paragraphs.paragraphs_type.journey_planner.yml:1]: The paragraph type must be defined in `hook_helfi_paragraph_types()` as well. `modules/custom/helfi_kymp_content/helfi_kymp_content.module` is probably good place for it.

See https://github.com/City-of-Helsinki/drupal-helfi-platform-config/blob/main/documentation/development.md#paragraph-types for documentation.

**Review by tuutti — CHANGES_REQUESTED** [review]

File: public/themes/custom/hdbt_subtheme/templates/paragraph/paragraph--journey-planner.html.twig [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      component_content_class: 'journey-planner journey-planner--embed',
+    }
+  %}
+    {% block component_content %}
+
+      <iframe 
+        width="500"
+        height="250"
+        style="border-radius: 10px;"
+        src="https://reittiopas.hsl.fi/haku?bikeOnly=1&lang={{ langcode }}"
```
[the remark this answers is not in the document]
Comment (xkhaven) [reply public/themes/custom/hdbt_subtheme/templates/paragraph/paragraph--journey-planner.html.twig]: Updated

File: public/themes/custom/hdbt_subtheme/templates/paragraph/paragraph--journey-planner.html.twig [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      component_classes: [ 'component--journey-planner' ],
+      component_title: content.field_title,
+      component_description: content.field_description,
+      component_content_class: 'journey-planner journey-planner--embed',
+    }
+  %}
+    {% block component_content %}
+
+      <iframe 
+        width="500"
```
[the remark this answers is not in the document]
Comment (xkhaven) [reply public/themes/custom/hdbt_subtheme/templates/paragraph/paragraph--journey-planner.html.twig]: Changed method of implementing iframe which fixes this.

File: public/themes/custom/hdbt_subtheme/hdbt_subtheme.theme [hunk 4]
```diff
@@ -96,3 +96,10 @@ function hdbt_subtheme_preprocess_node(array &$variables) {
     $variables['district_titles'] = implode(', ', $district_titles);
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function hdbt_subtheme_preprocess_paragraph__journey_planner(array &$variables) {
+  $variables['langcode'] = \Drupal::languageManager()->getCurrentLanguage()->getId();
+}
```
[the remark this answers is not in the document]
Comment (xkhaven) [reply public/themes/custom/hdbt_subtheme/hdbt_subtheme.theme]: Updated using as suggested above.

File: conf/cmi/paragraphs.paragraphs_type.journey_planner.yml (hunk 1 above)
Comment (xkhaven) [reply conf/cmi/paragraphs.paragraphs_type.journey_planner.yml:1]: Added to the suggest file.

**Review by teroelonen — APPROVED** [review]
This works well for me now 🦖

**Pull request merged**